Xamarin.Forms.Core/BindingExpression.cs

```diff
@@ -423,19 +423,28 @@ bool TryConvert(BindingExpressionPart part, ref object value, Type convertTo, bo
 				return true;
 
 			object original = value;
+			var locale = CultureInfo.InvariantCulture;
 			try
 			{
 				var stringValue = value as string ?? string.Empty;
-				// see: https://bugzilla.xamarin.com/show_bug.cgi?id=32871
-				// do not canonicalize "*.[.]"; "1." should not update bound BindableProperty
-				if (stringValue.EndsWith(".") && DecimalTypes.Contains(convertTo))
-					throw new FormatException();
 
-				// do not canonicalize "-0"; user will likely enter a period after "-0"
-				if (stringValue == "-0" && DecimalTypes.Contains(convertTo))
-					throw new FormatException();
+				if (DecimalTypes.Contains(convertTo))
+				{
+					// see: https://bugzilla.xamarin.com/show_bug.cgi?id=32871
+					// do not canonicalize "*.[.]"; "1." should not update bound BindableProperty
+					if (stringValue.EndsWith("."))
+						throw new FormatException();
+
+					// do not canonicalize "-0"; user will likely enter a period after "-0"
+					if (stringValue == "-0")
+						throw new FormatException();
+
+					// to change locale if stringValue contains decimal separator for that locale
+					if (stringValue.IndexOf(CultureInfo.CurrentCulture.NumberFormat.NumberDecimalSeparator) != -1)
```

Comment:
This should be removed; We shouldn't infer a culture by testing for culture specific values in the string. If we decide to allow non-invariant culture then we'll simply get the culture from the UI thread.